pkg/api/openshiftcluster.go

```diff
@@ -312,6 +312,57 @@ const (
+	VMSizeStandardD16asV4: {Type: VMSizeStandardD16asV4, CoreCount: 16, Family: "standardDASv4Family"},
+	VMSizeStandardD32asV4: {Type: VMSizeStandardD32asV4, CoreCount: 32, Family: "standardDASv4Family"},
+
+	VMSizeStandardD4sV3:  {Type: VMSizeStandardD4sV3, CoreCount: 4, Family: "standardDSv3Family"},
+	VMSizeStandardD8sV3:  {Type: VMSizeStandardD8sV3, CoreCount: 8, Family: "standardDSv3Family"},
+	VMSizeStandardD16sV3: {Type: VMSizeStandardD16sV3, CoreCount: 16, Family: "standardDSv3Family"},
+	VMSizeStandardD32sV3: {Type: VMSizeStandardD32sV3, CoreCount: 32, Family: "standardDSv3Family"},
+
+	VMSizeStandardE4sV3:   {Type: VMSizeStandardE4sV3, CoreCount: 4, Family: "standardESv3Family"},
+	VMSizeStandardE8sV3:   {Type: VMSizeStandardE8sV3, CoreCount: 8, Family: "standardESv3Family"},
+	VMSizeStandardE16sV3:  {Type: VMSizeStandardE16sV3, CoreCount: 16, Family: "standardESv3Family"},
+	VMSizeStandardE32sV3:  {Type: VMSizeStandardE32sV3, CoreCount: 32, Family: "standardESv3Family"},
+	VMSizeStandardE64isV3: {Type: VMSizeStandardE64isV3, CoreCount: 64, Family: "standardESv3Family"},
+	VMSizeStandardE64iV3:  {Type: VMSizeStandardE64iV3, CoreCount: 64, Family: "standardESv3Family"},
+
+	VMSizeStandardF4sV2:  {Type: VMSizeStandardF4sV2, CoreCount: 4, Family: "standardFSv2Family"},
+	VMSizeStandardF8sV2:  {Type: VMSizeStandardF8sV2, CoreCount: 8, Family: "standardFSv2Family"},
+	VMSizeStandardF16sV2: {Type: VMSizeStandardF16sV2, CoreCount: 16, Family: "standardFSv2Family"},
+	VMSizeStandardF32sV2: {Type: VMSizeStandardF32sV2, CoreCount: 32, Family: "standardFSv2Family"},
+	VMSizeStandardF72sV2: {Type: VMSizeStandardF72sV2, CoreCount: 72, Family: "standardFSv2Family"},
+
+	VMSizeStandardM128ms: {Type: VMSizeStandardM128ms, CoreCount: 128, Family: "standardMSFamily"},
+	VMSizeStandardG5:     {Type: VMSizeStandardG5, CoreCount: 32, Family: "standardGFamily"},
+	VMSizeStandardGS5:    {Type: VMSizeStandardGS5, CoreCount: 32, Family: "standardGFamily"},
+
+	VMSizeStandardL4s:    {Type: VMSizeStandardL4s, CoreCount: 4, Family: "standardLsv2Family"},
+	VMSizeStandardL8s:    {Type: VMSizeStandardL8s, CoreCount: 8, Family: "standardLsv2Family"},
+	VMSizeStandardL16s:   {Type: VMSizeStandardL16s, CoreCount: 16, Family: "standardLsv2Family"},
+	VMSizeStandardL32s:   {Type: VMSizeStandardL32s, CoreCount: 32, Family: "standardLsv2Family"},
+	VMSizeStandardL8sV2:  {Type: VMSizeStandardL8sV2, CoreCount: 8, Family: "standardLsv2Family"},
+	VMSizeStandardL16sV2: {Type: VMSizeStandardL16sV2, CoreCount: 16, Family: "standardLsv2Family"},
+	VMSizeStandardL32sV2: {Type: VMSizeStandardL32sV2, CoreCount: 32, Family: "standardLsv2Family"},
+	VMSizeStandardL48sV2: {Type: VMSizeStandardL48sV2, CoreCount: 48, Family: "standardLsv2Family"},
+	VMSizeStandardL64sV2: {Type: VMSizeStandardL64sV2, CoreCount: 64, Family: "standardLsv2Family"},
+}
+
+func VMFromTypeName(vmType VMSize) AzureVM {
+	return vmTypesMap[vmType]
+}
```

Comment:
Not sure that I see a point in this wrapper. Especially if we move the map to the validation code.

Reply:
This wrapper is to simulate a const, so we can't modify the map.